### PR TITLE
feat: automatic insertion of primary key tests

### LINKED ISSUE #194: [smart tests] automatically add tests for primary key
If a user adds the `primary_key` `{{config()}}` attribute we should automatically add unique ans not null tests.

We need to make sure the user knows we're adding and testing those for them though.

Probably we should also offer a CLI + sugar_config argument to disable those smart features as they may not fit all user's needs.

### PR DESCRIPTION
**Pull request by virvirlopez**
Resolves #194 

# Description

Automatic insertion of the primary key tests into the primary key column.

If an SQL file has a configuration specifying the primary key, this new functionality will try (is try because if the tests did not pass it will not add them) to add the unique test and the null test (only if there are not implemented already).

# How was the change tested
Ran it in local.

# Checklist

(Ideally, all boxes are checked by the time we merged the PR, if you don't know how to do any of these don't hesitate to say so in the PR and we'll help you out.)

- [X] I formatted my PR name according to [CONTRIBUTING.md](CONTRIBUTING.md)
- [x] I added a news fragment to help populating the changelog as encouraged in [CONTRIBUTING.md](CONTRIBUTING.md)
- [x] I added "Closes #<issue_number>" in the "Issue Information" section (if no issue, feel free to tick thick the box anyway).

### CODE REVIEW

File: tests/doc_task_test.py [hunk 1]
```diff
@@ -986,8 +986,147 @@ def test_move_name_and_description_to_first_position(content, result):
...
+                        "name": "testmodel",
+                        "description": "No description for this model.",
+                        "columns": [{"name": "id"}],
+                    }
+                ]
+            },
+            "testmodel",
+            "id",
+            False,
+            id="primary key tests are not presented",
```
Comment (bastienboutonnet) [inline tests/doc_task_test.py:1041]:
```suggestion
            id="primary key tests are not implemented",
```

File: tests/doc_task_test.py [hunk 2]
```diff
@@ -986,8 +986,147 @@ def test_move_name_and_description_to_first_position(content, result):
...
+                        "name": "testmodel",
+                        "description": "No description for this model.",
+                        "columns": [],
+                    }
+                ]
+            },
+            "testmodel",
+            "id",
+            None,
+            id="the primary key does not exists",
```
Comment (bastienboutonnet) [inline tests/doc_task_test.py:1071]:
```suggestion
            id="primary key not documented in schema.yml",
```

**Review by bastienboutonnet — COMMENTED** [review]
Really really cool @virvirlopez I've changed the base branch to `dev/0.1.0` just to put those new features aside for a bit while we finalise and wait for potential issues with the current release in case we have to ship more patches.

I made some suggestions and questions for you.

Our coverage is taking a big hit. Do you think you could add tests on the add function? I know it's "just" a runner but the logic inside is actually not simple and there are a few boolean logic pieces that I would like to make sure we have covered. What do you think?

Other than that, I think it's gonna work especially if you address my concern around how we get the path of the corresponding schema.yml file (see my comment in the code)

Finally, if you could:
- add "Resolves #ticket-numer in the PR description
- write a news fragment for the changelog that would be amazing.

Oh one more: make sure you check the issue and see if any if what I say or link too makes sense https://github.com/bitpicky/dbt-sugar/issues/194

**Review by bastienboutonnet — COMMENTED** [review]
Looks good but can you make sure that you add a changelog entry please (instructions here: https://github.com/bitpicky/dbt-sugar/blob/main/CONTRIBUTING.md#shout-out-your-hardwork-in-the-changelog) this is a big feature so we want to make sure people know about it

Other than that the code change is fine with me.

**Review by bastienboutonnet — APPROVED** [review]
Fantastic feature!
